llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -3802,7 +3802,7 @@ bool X86AsmParser::validateInstruction(MCInst &Inst, const OperandVector &Ops) {
     //    VFMULCPHZrr   Dest, Src1, Src2
     //    VFMULCPHZrrk  Dest, Dest, Mask, Src1, Src2
     //    VFMULCPHZrrkz Dest, Mask, Src1, Src2
-    for (unsigned i = TSFlags & X86II::EVEX_K ? 2 : 1;
```

Comment:
I know it's subjective, but the new code with two levels of parens decreases readability.

Reply:
The second level seems like a bug, it's not consistent with how `type identifier = expr ? expr : expr;` is handled in a block.